zio-http/shared/src/main/scala/zio/http/codec/QueryCodecs.scala

```diff
@@ -16,18 +16,37 @@
 
 package zio.http.codec
 import zio.stacktracer.TracingImplicits.disableAutoTrace
+import zio.{Chunk, NonEmptyChunk}
 
 private[codec] trait QueryCodecs {
+
   def query(name: String): QueryCodec[String] =
-    HttpCodec.Query(name, TextCodec.string)
+    toSingleValue(name, HttpCodec.Query(name, TextCodec.string))
 
   def queryBool(name: String): QueryCodec[Boolean] =
-    HttpCodec.Query(name, TextCodec.boolean)
+    toSingleValue(name, HttpCodec.Query(name, TextCodec.boolean))
 
   def queryInt(name: String): QueryCodec[Int] =
-    HttpCodec.Query(name, TextCodec.int)
+    toSingleValue(name, HttpCodec.Query(name, TextCodec.int))
 
   def queryTo[A](name: String)(implicit codec: TextCodec[A]): QueryCodec[A] =
+    toSingleValue(name, HttpCodec.Query(name, codec))
+
+  def queryMultiValue(name: String): QueryCodec[Chunk[String]] =
```

Comment:
please call these methods `queryAllXXX` so it maches the logical counterparts in `QueryParams`